The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/11663529498/job/32472160185) [(plain)](https://github.com/rust-lang/rust/commit/98ad987b7d827f5b7b0fe1fa856a910c50e1e725/checks/32472160185/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#13 2.669 Building wheels for collected packages: reuse
#13 2.670   Building wheel for reuse (pyproject.toml): started
#13 2.913   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 2.914   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132720 sha256=026f3bb0f1aa8090b861fd0a0939cb1a782396d84c8aab7875096557d637a0f6
#13 2.914   Stored in directory: /tmp/pip-ephem-wheel-cache-dqo21f69/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#13 2.917 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#13 3.314 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#13 3.314 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 3.827 Collecting virtualenv
#13 3.827 Collecting virtualenv
#13 3.872   Downloading virtualenv-20.27.1-py3-none-any.whl (3.1 MB)
#13 3.934      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.1/3.1 MB 52.0 MB/s eta 0:00:00
#13 3.988 Collecting platformdirs<5,>=3.9.1
#13 3.992   Downloading platformdirs-4.3.6-py3-none-any.whl (18 kB)
#13 4.028 Collecting filelock<4,>=3.12.2
#13 4.050 Collecting distlib<1,>=0.3.7
#13 4.053   Downloading distlib-0.3.9-py2.py3-none-any.whl (468 kB)
#13 4.061      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 80.9 MB/s eta 0:00:00
#13 4.061      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 80.9 MB/s eta 0:00:00
#13 4.139 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 4.312 Successfully installed distlib-0.3.9 filelock-3.16.1 platformdirs-4.3.6 virtualenv-20.27.1
#13 DONE 4.4s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      184256 kB
DirectMap2M:     8204288 kB
DirectMap1G:    10485760 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.05s
##[endgroup]
downloading https://static.rust-lang.org/dist/2024-10-16/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/2024-10-16/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt
---
Diff in /checkout/library/core/src/char/methods.rs:1821:
 #[unstable(feature = "char_internals", reason = "exposed only for libstd", issue = "none")]
 #[cfg_attr(
     bootstrap,
-    rustc_const_stable(feature = "const_char_encode_utf16", since = "CURRENT_RUSTC_VERSION"),
+    rustc_const_stable(feature = "const_char_encode_utf16", since = "CURRENT_RUSTC_VERSION")
 #[doc(hidden)]
 #[inline]
 #[inline]
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/library/core/src/char/methods.rs" "/checkout/library/core/src/char/decode.rs" "/checkout/library/core/src/char/mod.rs" "/checkout/library/core/src/borrow.rs" "/checkout/library/core/src/str/count.rs" "/checkout/library/core/src/str/validations.rs" "/checkout/library/core/src/str/converts.rs" "/checkout/library/core/src/str/pattern.rs" "/checkout/library/core/src/str/lossy.rs" "/checkout/library/core/src/str/traits.rs" "/checkout/library/core/src/str/iter.rs" "/checkout/library/core/src/str/error.rs" "/checkout/library/core/src/str/mod.rs" "/checkout/library/core/src/cmp.rs" "/checkout/library/core/src/marker.rs" "/checkout/library/core/src/io/borrowed_buf.rs" "/checkout/library/core/src/io/mod.rs" "/checkout/library/core/src/fmt/num.rs" "/checkout/library/core/src/fmt/rt.rs" "/checkout/library/core/src/fmt/nofloat.rs" "/checkout/library/core/src/fmt/float.rs" "/checkout/library/core/src/fmt/builders.rs" "/checkout/library/core/src/fmt/mod.rs" "/checkout/library/core/src/ub_checks.rs" "/checkout/library/core/src/clone.rs" "/checkout/library/core/src/escape.rs" "/checkout/library/core/src/primitive.rs" "/checkout/library/core/src/prelude/common.rs" "/checkout/library/core/src/prelude/mod.rs" "/checkout/library/core/src/unicode/printable.rs" "/checkout/library/core/src/unicode/unicode_data.rs" "/checkout/library/core/src/unicode/mod.rs" "/checkout/library/core/src/bool.rs" "/checkout/library/core/src/primitive_docs.rs" "/checkout/library/core/src/clone/uninit.rs" "/checkout/library/core/src/intrinsics.rs" "/checkout/library/core/src/pin.rs" "/checkout/library/core/src/intrinsics/mir.rs" "/checkout/library/core/src/intrinsics/simd.rs" "/checkout/library/core/src/task/ready.rs" "/checkout/library/core/src/task/wake.rs" "/checkout/library/core/src/task/poll.rs" "/checkout/library/core/src/task/mod.rs" "/checkout/library/core/src/arch.rs" "/checkout/library/core/src/num/bignum.rs" "/checkout/library/core/src/num/dec2flt/table.rs" "/checkout/library/core/src/num/dec2flt/fpu.rs" "/checkout/library/core/src/num/dec2flt/slow.rs" "/checkout/library/core/src/num/dec2flt/common.rs" "/checkout/library/core/src/num/dec2flt/parse.rs" "/checkout/library/core/src/num/dec2flt/float.rs" "/checkout/library/core/src/num/dec2flt/decimal.rs" "/checkout/library/core/src/num/dec2flt/number.rs" "/checkout/library/core/src/num/dec2flt/mod.rs" "/checkout/library/core/src/num/dec2flt/lemire.rs" "/checkout/library/core/src/num/f16.rs" "/checkout/library/core/src/num/overflow_panic.rs" "/checkout/library/core/src/num/shells/i64.rs" "/checkout/library/core/src/num/shells/int_macros.rs" "/checkout/library/core/src/num/shells/i128.rs" "/checkout/library/core/src/num/shells/u8.rs" "/checkout/library/core/src/num/shells/i8.rs" "/checkout/library/core/src/num/shells/u128.rs" "/checkout/library/core/src/char/convert.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Mon Nov  4 11:50:25 UTC 2024
  network time: Mon, 04 Nov 2024 11:50:25 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/library/core/src/char/methods.rs
+++ b/library/core/src/char/methods.rs
@@ -711,7 +711,7 @@ impl char {
     /// '𝕊'.encode_utf16(&mut b);
     /// ```
     #[stable(feature = "unicode_encode_char", since = "1.15.0")]
-    #[rustc_const_unstable(feature = "const_char_encode_utf16", issue = "130660")]
+    #[rustc_const_stable(feature = "const_char_encode_utf16", since = "CURRENT_RUSTC_VERSION")]
     #[inline]
     pub const fn encode_utf16(self, dst: &mut [u16]) -> &mut [u16] {
         encode_utf16_raw(self as u32, dst)
@@ -1819,7 +1819,10 @@ pub const fn encode_utf8_raw(code: u32, dst: &mut [u8]) -> &mut [u8] {
 /// Panics if the buffer is not large enough.
 /// A buffer of length 2 is large enough to encode any `char`.
 #[unstable(feature = "char_internals", reason = "exposed only for libstd", issue = "none")]
-#[rustc_const_unstable(feature = "const_char_encode_utf16", issue = "130660")]
+#[cfg_attr(
+    bootstrap,
+    rustc_const_stable(feature = "const_char_encode_utf16", since = "CURRENT_RUSTC_VERSION")
+)]
 #[doc(hidden)]
 #[inline]
 pub const fn encode_utf16_raw(mut code: u32, dst: &mut [u16]) -> &mut [u16] {

--- a/library/core/src/lib.rs
+++ b/library/core/src/lib.rs
@@ -114,7 +114,6 @@
 #![feature(const_align_of_val_raw)]
 #![feature(const_alloc_layout)]
 #![feature(const_black_box)]
-#![feature(const_char_encode_utf16)]
 #![feature(const_eval_select)]
 #![feature(const_exact_div)]
 #![feature(const_float_methods)]

